src/extras/SceneUtils.js

```diff
@@ -18,21 +18,46 @@ THREE.SceneUtils = {
 
 	},
 
-	detach: function ( child, parent, scene ) {
+	detach: function ( child, scene ) {
+
+		var topParent = child.parent;
+		
+		while ( topParent.parent !== undefined ) {
+			
+			topParent = topParent.parent;
+			
+		}
+		
+		topParent.updateMatrixWorld( true );
```

Comment:
This updates the complete scene. Wouldn't it be better if we had a method that calculates the matrixWorld for only 1 parent chain??